src/Validation/AbstractValidation.php

```diff
@@ -75,9 +77,70 @@ public function isFileMinSizeValid(int $fileSize, int $minFileSize): bool
 	 */
 	public function isFileTypeValid(string $fileName, string $fileTypes): bool
 	{
-		$fileExtension = explode('.', $fileName);
-		$validTypes = explode(',', str_replace(' ', '', str_replace('.', '', $fileTypes)));
+		$validTypes = $this->parseFiletypesString($fileTypes);
+
+		return in_array($this->getFileExtensionFromFilename($fileName), $validTypes, true);
```

Comment:
If `$validTypes` is a list (from the documentation it looks like it is), you can do this instead

```suggestion
		return isset($validTypes[$this->getFileExtensionFromFilename($fileName)]);
```

It's more performant than using `in_array` 🙂

Reply:
Huh, `$validTypes` is an `array<string>`, e.g. `["pdf", "gif", "jpg"]`:

```
php > $a = ["pdf", "gif", "jpg"];
php > var_dump($a);
array(3) {
  [0]=>
  string(3) "pdf"
  [1]=>
  string(3) "gif"
  [2]=>
  string(3) "jpg"
}
php > var_dump(isset($a["pdf"]));
bool(false)
php > var_dump(in_array("pdf", $a, true));
bool(true)
```

This would work if we used `array_flip` first, but this comes at a performance penalty: `array_flip` has a complexity of O(n), as does `isset` (although it's very fast, it's still linear complexity). 

`in_array` (which is O(n) as well) is slower than `isset`, but faster than `array_flip` + `isset` in benchmarks: https://3v4l.org/c85vN vs. https://3v4l.org/JWD7Q - it's a negligible difference, but one that's in favor of `in_array`. 

I'm all ears for microoptimizations, though 😄

Reply:
Forgot about the flip part 🙈 

This part: 

> has a complexity of O(n), as does isset (although it's very fast, it's still linear complexity).

is not true, because `isset` is O(1) (it just accesses the array to check for existence, while `in_array` is O(n) because it actually iterates over the array).

It's still a microoptimisation part since `in_array` is opcode cached function so it's ok to use it.